src/mds/InoTable.cc

```diff
@@ -190,3 +190,22 @@ void InoTable::generate_test_instances(list<InoTable*>& ls)
 {
   ls.push_back(new InoTable());
 }
+
+
+bool InoTable::repair_inotable(inodeno_t id)
+{
+	dout(0) << "repair inotable " << id << dendl;
+	if (free.contains(id)) {
```

Comment:
This check is backwards; repair needs doing if the inode number (in use by the inode we are checking) is marked as free.  We should skip if the inode is marked as *not* free (i.e. correct)